codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -404,12 +415,12 @@ private void applyLocalSettings(Http2Settings settings) throws Http2Exception {
 
             Long headerTableSize = settings.headerTableSize();
             if (headerTableSize != null) {
-                headerTable.maxHeaderTableSize(headerTableSize);
+                headerConfig.maxHeaderTableSize(headerTableSize);
             }
 
             Long maxHeaderListSize = settings.maxHeaderListSize();
             if (maxHeaderListSize != null) {
-                headerTable.maxHeaderListSize(maxHeaderListSize);
+                headerConfig.maxHeaderListSize(maxHeaderListSize, maxHeaderListSizeGoAway(maxHeaderListSize));
```

Comment:
Just so I understand ... after the remote peer acks our settings, we come here and set the goAway threshold to `1.5 * maxHeadersListSize`? After the ack is received, shouldn't it just be `maxHeadersListSize`?